syntax = "proto3";

package overlay;

service Windows {
	rpc CreateWindowGroup (CreateWindowGroupRequest) returns (CreateWindowGroupResponse) {}
	rpc CreateWindowInGroup (CreateWindowRequest) returns (CreateWindowResponse) {}
	rpc BufferForWindow (BufferForWindowRequest) returns (BufferForWindowResponse) {}
}

message WindowGroupProperties {
	sint32 z = 1;
	double opacity = 2;
	bool hidden = 3;
}

message WindowProperties {
	uint64 width = 1;
	uint64 height = 2;
	uint64 x = 3;
	uint64 y = 4;
	sint32 z = 5;
	double opacity = 6;
	bool hidden = 7;
}

message CreateWindowGroupRequest {
	WindowGroupProperties properties = 1;
}

message CreateWindowGroupResponse {
	bytes id = 1;
}

message CreateWindowRequest {
	bytes group_id = 1;
	WindowProperties properties = 2;
}

message CreateWindowResponse {
	bytes id = 1;
}

message BufferForWindowRequest {
	bytes group_id = 1;
	bytes window_id = 2;
	bytes buffer = 3;
}

message BufferForWindowResponse {

}